policy-controller/k8s/index/src/outbound/index.rs

```diff
@@ -303,6 +317,27 @@ impl Namespace {
         }
     }
 
+    fn reindex_services(&mut self, service_info: &HashMap<ServiceRef, ServiceInfo>) {
+        for routes in self.service_port_routes.values_mut() {
+            for routes in routes.watches_by_ns.values_mut() {
+                for route in routes.routes.values_mut() {
+                    for rule in route.rules.iter_mut() {
+                        for backend in rule.backends.iter_mut() {
```

Comment:
Just out of curiosity, I assume there's effectively no better way to accomplish what needs doing here without nesting for loops this deeply, is there?

Reply:
You're certainly right about clarity 😂. Any idea if there's a meaningful difference in the compiled binary? It _absolutely_ should't be a blocker to merging, by all means "looks good, ship it"-stamped. Just curious if it could be avoided or if maybe clang will take care of it either way.